### PR TITLE
Ensure prefetch_size and prefetch_count are 0 or greater

### LINKED ISSUE #944: BlockingChannel.basic_qos hanging on non-integer parameters
Under Python 3.6.4 and Pika 0.11.2, the `BlockingChannel.basic_qos` method hangs when setting its `prefetch_size` or `prefetch_count` parameter to a non-integer value.

Input:

```
$ python3 <<EOF
import pika
with pika.BlockingConnection() as connection:
    channel = connection.channel()
    channel.basic_qos(prefetch_size="hello")
EOF
```

### PR DESCRIPTION
**Pull request by lukebakken**
Fixes #944

### CODE REVIEW

File: tests/unit/channel_tests.py [hunk 1]
```diff
@@ -393,6 +393,18 @@ def test_basic_qos_raises_channel_closed(self):
...
+        self.assertRaises(ValueError, self.obj.basic_qos, None,
+                          'foo', 123, False)
+        self.assertRaises(ValueError, self.obj.basic_qos, None,
+                          -1, 123, False)
+
+    def test_basic_qos_raises_prefetch_count_value_error(self):
+        self.assertRaises(ValueError, self.obj.basic_qos, None,
+                          123, 'foo', False)
+        self.assertRaises(ValueError, self.obj.basic_qos, None,
+                          123, -1, False)
```
Comment (vitaly-krugl) [inline tests/unit/channel_tests.py]: In both of these tests, you need to validate that the parameters of the exception are what you expected not just the exception type. See example in https://github.com/pika/pika/blob/c0d0b3a2a69bdf394cd20d367a10a375e716e7b0/tests/unit/blocking_connection_tests.py#L149-L152. And you don't have to verify the verbatim exception text, but just some substring, such as `self.assertIn('prefetch_count', cm.exception.args[0])` (`ValueError` will have just the first string arg)

File: pika/channel.py [hunk 2]
```diff
@@ -457,6 +457,8 @@ def basic_qos(self,
         :param bool all_channels: Should the QoS apply to all channels
```
Comment (vitaly-krugl) [inline pika/channel.py]: Unrelated to this PR, but might be good to clarify in "The prefetch-count is ignored if the no-ack option is set." above that this applies to consumers. Perhaps word it as follows: "The prefetch-count is ignored by consumers with the no-ack option turned on."

**Comment by lukebakken** [comment]
@vitaly-krugl - this should be all set. I will address #951 next.

**Comment by lukebakken** [comment]
@vitaly-krugl - just FYI, this is ready for your review. Thanks!

File: tests/unit/channel_tests.py [hunk 3]
```diff
@@ -392,6 +392,26 @@ def test_basic_qos_raises_channel_closed(self):
         self.assertRaises(exceptions.ChannelClosed, self.obj.basic_qos, 0,
                           False, True)
 
+    def test_basic_qos_raises_prefetch_size_raises_error(self):
```
Comment (vitaly-krugl) [inline tests/unit/channel_tests.py]: Something doesn't look right with the method name - the word "raises" before "_prefetch_size". Did you mean to name it `test_basic_qos_*invalid*_prefetch_size_raises_error`?

File: tests/unit/channel_tests.py [hunk 4]
```diff
@@ -392,6 +392,26 @@ def test_basic_qos_raises_channel_closed(self):
...
+        self.obj._set_state(self.obj.OPEN)
+        with self.assertRaises(ValueError) as ex:
+            self.obj.basic_qos('foo', 123)
+        self.assertEqual("invalid literal for int() with base 10: 'foo'",
+                         ex.exception.args[0])
+        with self.assertRaises(ValueError) as ex:
+            self.obj.basic_qos(-1, 123)
+        self.assertIn('prefetch_size', ex.exception.args[0])
+
+    def test_basic_qos_raises_prefetch_count_raises_error(self):
```
Comment (vitaly-krugl) [inline tests/unit/channel_tests.py]: Something doesn't look right with the method name - the word "raises" before "_prefetch_count". Did you mean to name it `test_basic_qos_*invalid*_prefetch_count_raises_error`?

File: tests/unit/channel_tests.py (hunk 3 above)
Comment (lukebakken) [reply tests/unit/channel_tests.py]: Sure did, good catch

**Pull request merged**